server/server.go

```diff
@@ -1208,9 +1208,11 @@ KELoop:
 SRLoop:
 	for i, c := range s.clients {
 		// Recover shared secrets
-		b[i].srKP, b[i].dcKP = dcnet.SharedKeys(b[i].kx, ecdh, s.sid, s.msize,
+		b[i].srKP, b[i].dcKP, err = dcnet.SharedKeys(b[i].kx, ecdh, s.sid, s.msize,
 			s.run, starts[i], c.pr.MessageCount)
-
+		if err != nil {
+			log.Printf("%v: SharedKeys failed: %v", c.raddr(), err)
```

Comment:
blame assignment and this mix session have to be aborted entirely if this error is hit, there's no way to blame which peer is being malicious.